src/components/Api.js

```diff
@@ -0,0 +1,105 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    (this._baseUrl = baseUrl), (this._headers = headers);
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._checkResponse);
+  }
+
+  getUserInformation() {
+    return this._request(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).finally(() => {
+      console.log("Done with user info");
+    });
+  }
+
+  getInitialCards() {
+    return this._request(`${this._baseUrl}/cards`, { headers: this._headers })
+      .then((result) => {
+        return result;
+      })
+      .finally(() => {
+        console.log("Done with initial cards");
+      });
+  }
+  // User Profile
+  editProfileInformation({ name, about }) {
+    return this._request(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        about,
+      }),
+    })
+      .then((response) => {
+        return response;
+      })
+      .finally(() => {
+        console.log("Done sending User Information");
+      });
+  }
+  // Avatar
+  avatarInformation({ avatar }) {
+    return this._request(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar,
+      }),
+    });
+  }
+
+  // Adds New Card and sends to the server
+  addNewCardInformation({ name, link }) {
+    return this._request(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        link,
+      }),
+    })
+      .then((response) => {
+        return response;
+      })
+      .finally(() => {
+        console.log("Done adding New Card Information from Server");
+      });
+  }
+  // Delete card
+  deleteCardInformation(cardId) {
+    return this._request(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    })
+      .then((res) => {
+        return res;
+      })
+      .finally(() => {
+        console.log("Done deleting card");
+      });
+  }
+  // Add and Remove Like
+  likesAddInformation(cardId) {
```

Comment:
The name of the function should begin with a verb, reflecting the action that the function performs. Next, the elements to which the action refers are described. `addLike` would be good.